lxc/remote.go

```diff
@@ -449,22 +449,32 @@ func (c *cmdRemoteAdd) run(cmd *cobra.Command, args []string) error {
 
 	// Handle certificate prompt
 	if certificate != nil {
+		// Prompt for certificate acceptance if user did not allow us to blindly
+		// accept the remote certificate.
 		if !c.flagAcceptCert {
 			digest := shared.CertFingerprint(certificate)
 
 			fmt.Printf("%s: %s\n", i18n.G("Certificate fingerprint"), digest)
 			fmt.Print(i18n.G("ok (y/n/[fingerprint])?") + " ")
-			line, err := shared.ReadStdin()
-			if err != nil {
-				return err
-			}
+			for {
+				line, err := shared.ReadStdin()
+				if err != nil {
+					return err
+				}
 
-			if string(line) != digest {
+				// Continue with adding the remote if digest matches, or the user
+				// confirmed a fingerprint.
+				if string(line) == digest || strings.ToLower(string(line[0])) == i18n.G("y") {
```

Comment:
I wonder if we should still error if the input looks like a fingerprint but it doesn't match so that it doesn't block forever in a scripted scenario. E.g:
```
echo "${expectedFingerprint}" | lxc remote add <token>
```

Reply:
Thanks for suggestion. This is fixed now - if provided fingerprint length matches the remote certificate fingerprint length, but they are not equal, an error is immediately returned. 
```
$ lxc remote add 127.0.0.1
Certificate fingerprint: 99303272e5e88ed5ddc69d54b12761ab45b8afcece5e1262bed46e7c3e145203
ok (y/n/[fingerprint])? test
Please type 'y', 'n' or the fingerprint:test
Please type 'y', 'n' or the fingerprint:99303272e5e88ed5ddc69d54b12761ab45b8afcece5e1262bed46e7c3e14520    # 1 character shorter
Please type 'y', 'n' or the fingerprint:99303272e5e88ed5ddc69d54b12761ab45b8afcece5e1262bed46e7c3e145203x  # 1 character longer
Please type 'y', 'n' or the fingerprint:99303272e5e88ed5ddc69d54b12761ab45b8afcece5e1262bed46e7c3e14520x   # Lengths are equal, but the token is mismatched.
Error: Provided fingerprint does not match the certificate fingerprint
```

Reply:
In the above, when re-asking for the correct input, could you please add a space after the `:` in the prompt? Like that:
```
Please type 'y', 'n' or the fingerprint: test
```